### DISCUSSION TITLE
Back and Forward buttons

### DISCUSSION THREAD

**Discussion by gyll** [sumatrapdfreader/sumatrapdf#2614] [Ideas]
Hi there,

This is an absolutely GREAT reader, but it lacks the omnipresent Back and Forward BUTTONS in just about any other PDF reader i've ever used. I know there is the del/shiftDel (or alt<- / alt->) keys, but since a PDF reader is mostly used w/o a keyboard these two buttons are an absolute must-have on the toolbar. As far as i am concerned, this is a show stopper, and it is the ONLY showstopper for me to switch to sumatra. Can you please, please, please add these two buttons on the toolbar?

**Comment by kgurr** [comment]
There are extensive PDF files, with many internal links. In such files, back / forward buttons would be very helpful!

It would be conceivable to optionally activate / deactivate these buttons via the program settings.

**Comment by kjk** [comment]
If it'll happen, it'll be as part of making toolbar customizable, which is tracked by https://github.com/sumatrapdfreader/sumatrapdf/issues/1956 and you can vote on that feature at https://sumatrapdf.canny.io/admin/feedback/feature-requests/p/toolbar-customizability?boards=feature-requests

**Reply by gyll** [reply]
No, PLEASE, DON'T WAIT until you make the toolbar customizable, that may take a loooong time as it's not that trivial to implement, just add these buttons on the toolbar as it is now. IMHO, i really see no need for the next/previous page buttons at all, but at the very least, please just put in balance the usefulness of moving your cursor up on the toolbar, clicking a button, and then moving the cursor down inside the document vs giving your mouse wheel a spin - compared to the usefulness of being able to go back to what your were reading after clicking a link that will throw you God-knows-where in the document (not to mention multiple flow breaking clicks on successive links). I mean, if any of the two sets (next/prev page vs back/forward) should be present on the toolbar, it's the back/forward buttons that should be there by far from a usage statistics pov

**Reply by GitHubRulesOK** [reply]
Many users will fork/custom set the right / left button different, but that requires the individual to make very simple changes in the source definition of page forward and back icon actions, not quite as simple as swap keyboard commands 1:1 in the compiled versions.

**Comment by maxchendt** [comment]
please make it

and close this issue on Nov 9, 2016: https://github.com/sumatrapdfreader/sumatrapdf/issues/657
